tests/geminals/APIGGeminalCoefficients_test.cpp

```diff
@@ -0,0 +1,177 @@
+    BOOST_CHECK_THROW(GQCP::APIGGeminalCoefficients (4, 4), std::invalid_argument);
+}
+
+
+BOOST_AUTO_TEST_CASE ( constructor_vector ) {
+
+    // Check a correct constructor
+    Eigen::VectorXd g = Eigen::VectorXd::Zero(10);
+    BOOST_CHECK_NO_THROW(GQCP::APIGGeminalCoefficients (g, 2, 5));
+
+    // Check wrong parameters N_P and K
+    BOOST_CHECK_THROW(GQCP::APIGGeminalCoefficients (g, 2, 4), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::APIGGeminalCoefficients (g, 1, 5), std::invalid_argument);
+}
+
+
+BOOST_AUTO_TEST_CASE ( asMatrix ) {
+
+    // For N_P=2 and K=5, we have an APIG geminal coefficient matrix that looks like the following matrix:
+    Eigen::MatrixXd G (2, 5);
+    G << 1, 2, 3, 4, 5,
+         6, 7, 8, 9, 10;
+
+    // The geminal coefficients, arranged in a vector, are then represented by the following vector:
+    Eigen::VectorXd g (10);
+    g << 1, 2, 3, 4, 5, 6, 7, 8, 9, 10;
+
+
+    GQCP::APIGGeminalCoefficients gem_coeff (g, 2, 5);
+    BOOST_CHECK(gem_coeff.asMatrix().isApprox(G));
+
+    GQCP::APIGGeminalCoefficients gem_coeff_matrix (G);
+    BOOST_CHECK(gem_coeff_matrix.asVector().isApprox(g));
+}
+
+
+BOOST_AUTO_TEST_CASE ( vectorIndex ) {
+
+    size_t K = 5;
+    size_t N_P = 2;
+    GQCP::APIGGeminalCoefficients gem_coeff (N_P, K);
+
+
+    BOOST_CHECK_EQUAL(gem_coeff.vectorIndex(0, 2), 2);
+    BOOST_CHECK_EQUAL(gem_coeff.vectorIndex(1, 1), 6);
+    BOOST_CHECK_EQUAL(gem_coeff.vectorIndex(1, 2), 7);
+
+    // Require a throw if i > N_P
+    BOOST_REQUIRE_THROW(gem_coeff.vectorIndex(3, 3), std::invalid_argument);
+}
+
+
+BOOST_AUTO_TEST_CASE ( matrixIndex ) {
+
+    size_t K = 5;
+    size_t N_P = 2;
+    GQCP::APIGGeminalCoefficients gem_coeff (N_P, K);
+
+
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMajor(2), 0);
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMajor(6), 1);
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMajor(7), 1);
+
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMinor(2), 2);
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMinor(6), 1);
+    BOOST_CHECK_EQUAL(gem_coeff.matrixIndexMinor(7), 2);
+}
+
+
+BOOST_AUTO_TEST_CASE ( operator_call ) {
+    // Make an example for geminal coefficients for N_P=2 and K=5
+    //      1 2 3 4 5
+    //      6 7 8 9 10
+    Eigen::VectorXd g (10);
+    g << 1, 2, 3, 4, 5, 6, 7, 8, 9, 10;
+    GQCP::APIGGeminalCoefficients gem_coeff (g, 2, 5);
+
+
+    BOOST_CHECK_EQUAL(gem_coeff(2), 3);
+    BOOST_CHECK_EQUAL(gem_coeff(6), 7);
+    BOOST_CHECK_EQUAL(gem_coeff(7), 8);
+
+    BOOST_CHECK_EQUAL(gem_coeff(0, 2), 3);
+    BOOST_CHECK_EQUAL(gem_coeff(1, 1), 7);
+    BOOST_CHECK_EQUAL(gem_coeff(1, 2), 8);
+}
+
+
+BOOST_AUTO_TEST_CASE ( toWaveFunction_example1 ) {
+
+    size_t K = 3;
+    size_t N_P = 1;
+
+    Eigen::VectorXd g (3);
+    g << 1, 2, 3;
+    GQCP::APIGGeminalCoefficients gem_coeff (g, N_P, K);
+
+    Eigen::VectorXd ref_coefficients (3);
+    ref_coefficients << 1, 2, 3;
+
+    GQCP::FockSpace fock_space (K, N_P);
```

Comment:
You could add a function to `BaseAPIGGeminalCoefficients" that returns a FockSpace, rather having to re-enter the your constructor arguments in both FockSpace and Geminal class.

Reply:
That's a good suggestion. However, I feel like this would be just a workaround since we don't have the `toWaveFunction()`-issue cleared up yet.